stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -236,6 +252,93 @@ public class MessageListViewModel(
         }
     }
 
+    /**
+     * Counts how many messages the user hasn't read already. This is based on what the last message they've seen is,
+     * and the current message state.
+     *
+     * @param newMessageState - The state that tells us if there are new messages in the list.
+     * @return - [Int] which describes how many messages come after the last message we've seen in the list.
+     * */
+    private fun getUnreadMessageCount(newMessageState: NewMessageState? = currentMessagesState.newMessageState): Int {
+        if (newMessageState == null || newMessageState == MyOwn) return 0
+
+        val messageItems = currentMessagesState.messageItems
+        val lastSeenMessagePosition =
+            getLastSeenMessagePosition(if (isInThread) lastSeenThreadMessage else lastSeenChannelMessage)
+        var unreadCount = 0
+
+        for (i in 0..lastSeenMessagePosition) {
+            val messageItem = messageItems[i]
+
+            if (!messageItem.isMine && messageItem.message.deletedAt == null) {
+                unreadCount++
+            }
+        }
+
+        return unreadCount
+    }
+
+    /**
+     * Gets the list position of the last seen message in the list.
+     *
+     * @param lastSeenMessage - The last message we saw in the list.
+     * @return - [Int] list position of the last message we've seen.
+     * */
+    private fun getLastSeenMessagePosition(lastSeenMessage: MessageItem?): Int {
+        if (lastSeenMessage == null) return 0
```

Comment:
+1 here const imo for lastSeenMessagePosition = 0 can be declared

Please ignore this comment if you feel otherwise

Reply:
As per the comments above, I think this is another case where it's pretty clear what we're doing. If there is no lastSeenMessage, then its position is `0`, or we're at the start of the list.

We could return -1 here or something potentially, but that would require extra checks for that specific case.